shared/src/balancer/event_handler.rs

```diff
@@ -120,22 +124,22 @@ impl PoolSpecialization {
     }
 }
 
-impl BalancerPools {
-    fn try_upgrade(&mut self) {
+impl PoolRegistry {
+    fn try_upgrade(&mut self) -> Result<()> {
```

Comment:
nit: maybe some docs on when this should error?

Reply:
Will address and test this in #696.